react/react1/week2/vite-project/src/components/TodoList/TodoList.jsx

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable react/prop-types */
+import React from "react";
+import { useState } from "react";
+import TodoItem from "./todoItem/todoItem";
+
+function TodoList(props) {
+    const [data, setData] = useState(props.todos);
+    const [nextId, setNextId] = useState(props.todos.length + 1);
+
+    const handleClick = () => {
+        setNextId(nextId + 1);
+        const newItem = { description: 'New task', id: nextId };
+        setData([...data, newItem]);
```

Comment:
Id rather you have the todo's as a state in the app component since this is where the todos are initialised. In that case we don't need any states in this component, and can just call a setDate prop function instead. Write me if this needs more explaining

Reply:
> Id rather you have the todo's as a state in the app component since this is where the todos are initialised. In that case we don't need any states in this component, and can just call a setDate prop function instead. Write me if this needs more explaining

Also, for this comment, is it possible to. move the todo array into TodoList component?